perf/reduce.jl

```diff
@@ -12,7 +12,7 @@ D = DataArray(A)
 E = DataArray(A, B)
 
 f(x) = 5 * x
-f{T<:Number}(x::Nullable{T}) = Nullable(5 * x.value, x.isnull)
+f{T<:Number}(x::Nullable{T}) = ifelse(isnull(x), Nullable{typeof(5*x.value)}, Nullable(5 * x.value))
```

Comment:
Yes, at least if we want to ensure type stability in all cases. But I just realized this is missing `()`, and the fact that tests haven't failed means that it's not really tested on nulls. Will fix.